sympy/matrices/common.py

```diff
@@ -2118,6 +2118,10 @@ def subs(self, *args, **kwargs):  # should mirror core.basic.subs
         >>> Matrix(_).subs(y, x)
         Matrix([[x]])
         """
+
+        if isinstance(args[0], zip):
+            args = (list(args[0]),)
```

Comment:
There are other possibilities for what args[0] could be that would have this problem e.g. map etc (athough zip is most common).

I think it would be better to do something like convert args to a tuple if whenever it is not a tuple.

Also this would ignore args[1] etc if supplied.

Reply:
Hey @oscarbenjamin, thanks for the feedback. In all honesty, I didn't realize that creating a fork of the repository would still notify contributors of any pull requests I made...rookie mistake! I tried inputs on maps and those seemed to work. I'll now try to enhance this pull request like you suggested and also add test cases as well as some release notes.

Reply:
Also, would you recommend that I create a completely new PR?

Reply:
> Or can I just modify this one?

Modifying this one should do.